### PR TITLE
super admin side bar component

### LINKED ISSUE #52: [FEAT] Super Admin Side Bar Component
### Description:

Implement the Super Admin Side Nav Bar using ShadCN UI components. The Nav Bar should adhere strictly to the design specifications provided. Below are the detailed instructions and requirements.

**Requirements:**

1. **UI Components:**
   - Use ShadCN UI components for consistency.
   - Ensure components match the design specifications provided.

2. **Design Specifications:**
   - Refer to the provided design for dimensions, colors, and layout.
   - Include all navigation tabs as designed Dashboard, Products, Users, Email Template, Squeeze pages, Waitlist Pages, Settings 

3. **Functionality:**
   - Make sure all navigation tabs route to their specified pages 

4. **Accessibility:**
   - Ensure all components are accessible and follow best practices for ARIA roles and attributes.

    
**Acceptance Criteria:**

- [ ] The `SuperAdminSideNavBar` component is created and located in the `components` directory.
- [ ] The Side Nav Bar matches the design specifications.
- [ ] The Side Nav Bar functions correctly and includes all items as per the design.
- [ ] Components are accessible and follow best practices.

**Attachments:**
Screenshot 2024-07-17 at 13 36 23

[Link to Design File](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2179-42622&m=dev)

- [ShadCN UI Documentation](https://ui.shadcn.com/)

### PR DESCRIPTION
**Pull request by wale2009**
# Description

This pull request introduces a new component, SuperAdminSideNavBar.tsx, which implements a functional side navigation bar for Super Admin users. The navbar includes:

- Necessary icons for navigation elements.
- A function to navigate to the Settings page from the navbar.
- Links for various functionalities accessible through button clicks.

**Closes #52 **

# Changes proposed

## What were you told to do?

 Super Admin Side Bar Component 

1. UI Components: 
- Use ShadCN UI components for consistency.
- Ensure components match the design specifications provided.
2. Design Specifications: 
- Refer to the provided design for dimensions, colors, and layout.
-  Include all navigation tabs as designed Dashboard, Products, Users, Email Template, Squeeze pages, Waitlist Pages, Settings
3. Functionality:
-  Make sure all navigation tabs route to their specified pages
4. Accessibility:
- Ensure all components are accessible and follow best practices for ARIA roles and attributes.

## What did you do?

- **New Component:** I  created a new file named app/component/SuperAdminSideNavBar.tsx which contains the code for the side navigation bar component.

- **Functionality:** I designed the navbar with relevant icons for easy identification of navigation options. I implemented a function that allows Super Admins to navigate to a dedicated Settings page directly from the navbar. I created links for various functionalities within the navbar. These links are likely tied to buttons or menu items that, when clicked, trigger specific actions within the application.

- **Integration:** Modified the app/component/root.tsx file to integrate the newly created SuperAdminSideNavBar.tsx component. This likely involves adding the component to the application's layout or structure.

-  **Dependency**: I also updated the app/component/ui/accordion.tsx file to utilize an Accordion component from a dependency named Shandcn/ui. This suggests you might be using an accordion element within the side navbar for collapsible sections or menus.

**Figma Link:** 
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2179-42622&m=dev

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

 - [X] My code follows the code style of this project.
  - [X] This PR does not contain plagiarized content.
 - [X] The title and description of the PR is clear and explains the approach.
 - [X] I am making a pull request against the **dev branch** (left side).
 - [X] My commit messages styles matches our requested structure.
 - [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos
[
Side Nav Bar
](url)

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -25,9 +26,12 @@ export function Layout({ children }: { children: React.ReactNode }) {
         <Links />
       </head>
       <body>
```
Comment (chumex412) [inline app/root.tsx:28]: Why not create a file with the name super-admin in the app/routes ?

Please move this code to that file

File: app/components/SuperAdminSideNavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      path: "/admin/squeeze-pages",
+    },
+    {
+      name: "Waitlist Page",
+      icon: "/icons/waitlist.png",
+      path: "/admin/waitlist-page",
+    },
+    { name: "Settings", icon: "/icons/settings.png" },
+  ];
+
```
Comment (chumex412) [inline app/components/SuperAdminSideNavBar.tsx:36]: move this to another file wthin the data folder:

settingsMenuItems, organizationItem and mainMenuItems

**Review by chumex412 — CHANGES_REQUESTED** [review]

File: app/root.tsx (hunk 1 above)
Comment (wale2009) [reply app/root.tsx:29]: Sorry I don't understand

Comment (chumex412) [reply app/root.tsx:29]: Create a folder called data and inside it create a file called navigation. Move that array of navigation items to that file, export and import it into this file

**Review by chumex412 — APPROVED** [review]

**Comment by chumex412** [comment]
Please ensure to move any mapped array to a folder under app eg app/data/navigation.ts

**Pull request merged**